libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ProbabilityDistributionChart.tsx

```diff
@@ -0,0 +1,244 @@
+    if (this.state.probabilityOption === undefined) {
+      const probabilityOptions = this.getProbabilityOptions();
+      if (probabilityOptions.length > 0) {
+        this.setState({ probabilityOption: probabilityOptions[0] });
+      }
+    }
+  }
+
+  public render(): React.ReactNode {
+    const classNames = modelOverviewChartStyles();
+    if (!this.context.jointDataset.hasPredictedProbabilities) {
+      return;
+    }
+    const theme = getTheme();
+
+    const selectedDatasetCohorts = this.props.datasetCohorts.filter(
+      (_cohort, index) => {
+        return this.props.selectedDatasetCohorts.includes(index);
+      }
+    );
+    const selectedFeatureBasedCohorts = this.props.featureBasedCohorts.filter(
+      (_cohort, index) => {
+        return this.props.selectedFeatureBasedCohorts.includes(index);
+      }
+    );
+    const selectedCohorts = selectedDatasetCohorts.concat(
+      selectedFeatureBasedCohorts
+    );
+    const selectedCohortNames = selectedCohorts.map(
+      (cohort) => cohort.cohort.name
+    );
+
+    const probabilityOptions = this.getProbabilityOptions();
+
+    if (probabilityOptions.length === 0) {
+      return React.Fragment;
+    }
+
+    if (this.state.probabilityOption === undefined) {
+      this.setState({ probabilityOption: probabilityOptions[0] });
+      return React.Fragment;
+    }
+
+    const boxPlotData = selectedCohorts.map((cohort, index) => {
+      return calculateBoxPlotDataFromErrorCohort(
+        cohort,
+        index,
+        this.state.probabilityOption!.key.toString()
+      );
+    });
+    const outlierData = boxPlotData
+      .map((cohortBoxPlotData) => cohortBoxPlotData.outliers)
+      .map((outlierProbs, cohortIndex) => {
+        return outlierProbs.map((prob) => [cohortIndex, prob]);
+      })
+      .reduce((list1, list2) => list1.concat(list2), []);
+
+    const noCohortSelected =
+      this.props.selectedDatasetCohorts.length +
+        this.props.selectedFeatureBasedCohorts.length ===
+      0;
+
+    return (
+      <>
+        <Stack horizontal>
+          {!noCohortSelected && (
+            <Stack.Item className={classNames.verticalAxis}>
+              <DefaultButton
+                className={classNames.rotatedVerticalBox}
+                text={
+                  localization.ModelAssessment.ModelOverview
+                    .cohortSelectionButton
+                }
+                onClick={this.props.onChooseCohorts}
+              />
+            </Stack.Item>
+          )}
+          <Stack.Item className={classNames.chart}>
+            {noCohortSelected && (
+              <div className={classNames.placeholderText}>
+                <Text>
+                  {
+                    localization.ModelAssessment.ModelOverview
+                      .boxPlotPlaceholder
+                  }
+                </Text>
+              </div>
+            )}
+            {!noCohortSelected && (
+              <Stack>
+                <BasicHighChart
```

Comment:
This file looks quite large already, I wonder if it might help to refactor this "BasicHighChart" chart code out to a separate component.  Maybe it's not necessary now though, but in case more code gets added here I would recommend that.

Reply:
I will have another PR to add a different version of this chart (using lines rather than boxes). At that point I'll split it up because we'll certainly cross the 300 lines. For this PR I'd prefer to keep it as is because I don't quite know yet which parts will be common.